var/spack/repos/builtin/packages/halide/package.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Halide(CMakePackage, PythonExtension):
+    """Halide is a programming language designed to make it easier to write
+    high-performance image and array processing code on modern machines."""
+
+    homepage = "https://halide-lang.org/"
+    url = "https://github.com/halide/Halide/archive/refs/tags/v14.0.0.tar.gz"
+    git = "https://github.com/halide/Halide.git"
+    maintainers = ["wraith1995"]
+    version("main", branch="main")
+    version("14.0.0", sha256="f9fc9765217cbd10e3a3e3883a60fc8f2dbbeaac634b45c789577a8a87999a01")
+    variant(
+        "build_type",
+        default="Release",
+        description="The build type to build",
+        values=("Release", "Debug", "RelWithDebInfo"),
+    )
+    generator = "Ninja"
+    variant("python", default=False, description="Install python bindings")
+    variant("tutorials", default=False, description="Install the Halide Tutorials.")
+    variant("utils", default=False, description="Install the Halide Utilities.")
+    variant("tests", default=False, description="Build and Run Halide Tests and Apps.")
+    extends("python", when="+python")
+
+    _targets = " targets=arm,x86,nvptx,aarch64,hexagon,webassembly "
```

Comment:
These are _required_ to build halide or these are what you personally _prefer_?

Reply:
This is the suggestion in the readme and not based on my preferences. I don't know what will happen if you build without them.

Reply:
https://github.com/halide/Halide/tree/e8e1481633b4dd24bc0f0cf21459793ba66771b5#acquiring-llvm